### DISCUSSION TITLE
Does nuxt support parsing application/x-www-form-urlencoded body or just JSON?

### DISCUSSION THREAD

**Discussion by glopezep** [nuxt/nuxt#17902] [Questions] [answered]
.

**Comment by Aareksio** [comment, marked as answer]
Under nuxt 3 in node.js mode works the [h3 web server](https://github.com/unjs/h3). You can access the request body as [documented here](https://v3.nuxtjs.org/docs/directory-structure/server#accessing-req-data). The example suggests `useBody`, which as you noticed, parses JSON body. If you are expecting other content encoding, such as `application/x-www-form-urlencoded`, you must parse it accordingly.

Instead of `useBody`, you can access the request payload with [useRawBody](https://www.jsdocs.io/package/h3#useRawBody). The content of `application/x-www-form-urlencoded` is url encoded string - `param1=valueA&param2=valueB`. You can parse it using a query string parser of your choice. Popular packages include: [query-string](https://github.com/sindresorhus/query-string), [qs](https://github.com/ljharb/qs) or [ufo](https://github.com/unjs/ufo) if you nuxt hit you too hard.
